pkg/openstack/cinder.go

```diff
@@ -37,6 +38,20 @@ func ReconcileCinder(ctx context.Context, instance *corev1beta1.OpenStackControl
 			//cinder.Spec.DatabaseInstance = instance.Name // name of MariaDB we create here
 			cinder.Spec.DatabaseInstance = "openstack" //FIXME: see above
 		}
+		// if already defined in the top-level crd, we don't override
+		if len(cinder.Spec.ExtraMounts) == 0 {
+
+			var cinderVolumes []storage.CinderExtraVolMounts
+
+			for _, ev := range instance.Spec.ExtraMounts {
+				cinderVolumes = append(cinderVolumes, storage.CinderExtraVolMounts{
+					Name:      ev.Name,
+					Region:    ev.Region,
+					VolMounts: ev.VolMounts,
+				})
```

Comment:
I know that it was discussed in the lib-common change that the reason to keep a separate `*ExtraVolMounts` structs is to allow divergence between them so that different services can implement different features. However here we are copying data between the OpenstackExtraVolMount and CinderExtraVolMounts. I don't really see how this will keep working properly after Cinder struct starts diverging from the OpenStack struct.

Reply:
In this code we're actually building:

```
storage.CinderExtraVolMounts{
					Name:      ev.Name,
					Region:    ev.Region,
					VolMounts: ev.VolMounts,
				})
```

I see the `OpenStack` struct as a superset of this `Cinder` related struct, which means that even though I'm expecting different needs from different operators, I still see them getting a subset of what's added by the `OpenStack` one.
`Glance` can have different parameters from `Cinder`, and its `AppendVolume` function can diverge in the way volumes are filtered, but `OpenStack` here is a big structure that is used to define a common, `extraVolumes` definition that can be translated into each operator struct.

Reply:
Yes, I see your point of having a common, subset of input provided by `OpenStackExtraVolMount` that makes sense for each operator.
Thanks for the input, I'll follow up on this part too.